src/index.ts

```diff
@@ -130,51 +114,58 @@ const program = pipe(
       INPUT_SPECTRAL_RULESET,
     }) =>
       pipe(
-        getRepositoryInfoFromEvent(GITHUB_EVENT_PATH, INPUT_EVENT_NAME),
-        TE.chain(event =>
+        pipe(
+          getRepositoryInfoFromEvent(GITHUB_EVENT_PATH, INPUT_EVENT_NAME),
+          TE.map(event => ({ event }))
+        ),
+        TE.chain(({ event }) =>
+          pipe(
+            readFilesToAnalyze(INPUT_FILE_GLOB, GITHUB_WORKSPACE),
+            TE.chain(fileContents => createSpectralAnnotations(INPUT_SPECTRAL_RULESET, fileContents, GITHUB_WORKSPACE)),
+            TE.map(annotations => ({ event, annotations }))
+          )
+        ),
+        TE.chain(({ event, annotations }) =>
```

Comment:
Hmm I'd say this is going in the wrong direction. I'll probably have to check this out.